src/routes/frontdesk/+page.svelte

```diff
@@ -0,0 +1,118 @@
+<script lang="ts">
+  import { State, Order } from "$lib/types";
+  import OrderList from "./OrderList.svelte";
+  // TODO: instead of having an Order-type, interface directly with the DB using a svelte store
+  $: orders = [
+    new Order(123),
+    new Order(456, State.Complete),
+    new Order(789, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete),
+    new Order(999, State.Complete)
+  ];
+
+  export let data;
+  // Is undefined before anything is checked.
+  // Make sure to account for that when implementing logic based on it.
+  let selectedDrink;
+
+  let cart = [];
+  $: totalPrice = cart.reduce((sum, item) => sum + item.price, 0);
+  function addToCart() {
+    // Has to be re-assigned to trigger reactivity..
+    // https://svelte.dev/tutorial/svelte/deep-state
+    // ↑ does not seem to work here /shrug
+    cart = [...cart, selectedDrink];
+    // selectedDrink = undefined;
+  }
+
+  function removeFromCart(index) {
+    // Same as above :(
+    // cart.splice(index, 1);
+    // cart = [...cart];
+    cart = cart.slice(0, index).concat(cart.slice(index + 1));
+  }
+</script>
+
+<div class="grid h-full grid-cols-[3fr,auto,1.5fr,auto,1fr] grid-rows-[100%] gap-4">
+  <div class="flex flex-col overflow-y-auto">
+    {#each data.categories as category}
+      <div class="mb-8">
+        <h1 class="mb-4 text-2xl font-bold text-primary">{category.name}</h1>
+        <div class="grid grid-cols-1 gap-6 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4">
+          {#each category.expand.drinks_via_category as drink}
+            <label>
+              <input
+                type="radio"
+                name="drink"
+                class="peer hidden"
+                value={drink}
+                bind:group={selectedDrink}
+              />
+              <div
+                class="btn relative flex h-24 w-full flex-col items-center justify-center border-4 peer-checked:border-accent"
+              >
+                <span class="font-bold">{drink.name}</span>
+                <span class="absolute bottom-2 right-2 font-normal">{drink.price},-</span>
+              </div>
+            </label>
+          {/each}
+        </div>
+      </div>
+    {/each}
+  </div>
+
+  <div class="divider divider-horizontal"></div>
+
+  <div class="flex h-full flex-col justify-center gap-8">
+    <div class="h-1/2 overflow-y-auto">
+      <table class="table table-pin-rows table-auto list-none p-4 shadow-2xl">
```

Comment:
kan tabellen tas ut som et komponent på siden?

Reply:
Den bør kunne det, men problemet er at tabellen og valgene på venstre side må sende data mellom seg, så dette vil gjøre koden litt mer komplisert, derfor har jeg som en mvp skrevet all logikken i samme fil, som kan refactores senere til å være bedre organisert (kanskje etter #51 , når appending og slikt kan skrives om med deep reactivity og append / splice)

Reply:
Jeg er en gammel react-noob så jeg støtter et pragmatisk valg, for jeg ser svelte ikke gjør det like lett.

Det viktig det du sier da. Det å sende data mellom disse to komponentene, tvinger en til å være ærlig om hva data de forskjellige komponentene trenger. Dette holder kompleksitetsdemonene og javanissene ute av bildet.

Reply:
@IldenH jeg tagger deg og gir dere muligheten til å resolve uten endring.

Reply:
Jeg tror egentlig det kan gjøres ganske bra i svelte, ved å bruke bind, noe lignende `export let selectedItem` i `MenuSelection.svelte`, og `<MenuSelection bind:selectedItem={selectedItem} />`, der `let selectedItem;` er definert på starten av `+page.svelte`.  Jeg har bare valgt å ikke gjøre det med en gang.

Reply:
> Jeg tror egentlig det kan gjøres ganske bra i svelte

Sant, det har de nok hentet fra Vue, i og med at de også har bind.

React har inline components, noe som fjerner dette problemet helt. Da kan man definere et komponent i samme fil, som har den samme dataen. Og heller ekstrahere med dependencies når situasjonen tilsier at dette er lurt.

Hvis du kan, og svelte ikke deprecater bind i 5, så er det kjekt. Opp til deg, som sagt, inntil applikasjonen er i prod.